advanced/matrix/matrix.py

```diff
@@ -0,0 +1,328 @@
+from typing import Dict
+
+
+class Matrix:
+    def __init__(self, *args):
+        """
+        Initialize method - Organize the Matrix checks the input as well.
+        """
+        # Remove unnecessary brackets. Like so: (((1, 2), (3, 4)),)
+        if type(args[0][0]) == tuple:
+            self._matrix = args[0]
+        else:
+            self._matrix = args
+
+        self._check_input(self._matrix)
+
+    @staticmethod
+    def _check_input(matrix_value):
+        """
+        Check if the input is according to the format of squared tuple (n * n), for example: ((a, b), (c, d)).
+        Checks if a b c d are integers or floats.
+        Raise error otherwise.
+        """
+        format_is_ok = True
+        value_is_ok = True
+        len_mat = len(matrix_value)
+
+        for tup in matrix_value:
+            # Checks format (n * n matrix)
+            if len(tup) != len_mat:
+                format_is_ok = False
+
+            for value in tup:
+                # Checks the value of the tuples.
+                if not (type(value) == int or type(value) == float):
+                    value_is_ok = False
+
+        if not format_is_ok:
+            raise SyntaxError("Syntax error while defying the Matrix."
+                              " Needs to be a 'squared tuple' (n * n Matrix). => " + str(matrix_value))
+        if not value_is_ok:
+            raise TypeError("TypeError error while defying the Matrix."
+                            " All values must be float or integer numbers. => " + str(matrix_value))
+
+    def __str__(self):
+        """
+        Returns a string value of the Matrix class. For print() function.
+        """
+        return str(self._matrix)
+
+    def __repr__(self):
+        """
+        Returns representation of the matrix.
+        """
+        return "Matrix%s" % str(self._matrix)
+
+    def __len__(self):
+        """
+        :return: The length of the matrix. (The order n of the squared Matrix).
+        """
+        return len(self._matrix)
+
+    def __getitem__(self, item=None):
+        """
+        :param item: returns a specific item. Irrelevant in this case.
+        :return: The matrix object itself.
+        """
+        return self._matrix
+
+    def tuples(self):
```

Comment:
You declared a function - the user needs to call `x.tuples()`.
you should make this look like a property - so it be called as `x.tuples`